src/client/pythonEnvironments/main.ts

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {
+    CONDA_ENV_FILE_SERVICE,
+    CONDA_ENV_SERVICE,
+    CURRENT_PATH_SERVICE,
+    GLOBAL_VIRTUAL_ENV_SERVICE,
+    ICondaService,
+    IInterpreterLocatorHelper,
+    IInterpreterLocatorProgressService,
+    IInterpreterLocatorService,
+    IInterpreterWatcher,
+    IInterpreterWatcherBuilder,
+    IKnownSearchPathsForInterpreters,
+    INTERPRETER_LOCATOR_SERVICE,
+    IVirtualEnvironmentsSearchPathProvider,
+    KNOWN_PATH_SERVICE,
+    PIPENV_SERVICE,
+    WINDOWS_REGISTRY_SERVICE,
+    WORKSPACE_VIRTUAL_ENV_SERVICE
+} from '../interpreter/contracts';
+import { IPipEnvServiceHelper, IPythonInPathCommandProvider } from '../interpreter/locators/types';
+import { IServiceManager } from '../ioc/types';
+import { PythonInterpreterLocatorService } from './discovery/locators';
+import { InterpreterLocatorHelper } from './discovery/locators/helpers';
+import { InterpreterLocatorProgressService } from './discovery/locators/progressService';
+import { CondaEnvFileService } from './discovery/locators/services/condaEnvFileService';
+import { CondaEnvService } from './discovery/locators/services/condaEnvService';
+import { CondaService } from './discovery/locators/services/condaService';
+import { CurrentPathService, PythonInPathCommandProvider } from './discovery/locators/services/currentPathService';
+import {
+    GlobalVirtualEnvironmentsSearchPathProvider,
+    GlobalVirtualEnvService
+} from './discovery/locators/services/globalVirtualEnvService';
+import { InterpreterHashProvider } from './discovery/locators/services/hashProvider';
+import { InterpeterHashProviderFactory } from './discovery/locators/services/hashProviderFactory';
+import { InterpreterFilter } from './discovery/locators/services/interpreterFilter';
+import { InterpreterWatcherBuilder } from './discovery/locators/services/interpreterWatcherBuilder';
+import { KnownPathsService, KnownSearchPathsForInterpreters } from './discovery/locators/services/KnownPathsService';
+import { PipEnvService } from './discovery/locators/services/pipEnvService';
+import { PipEnvServiceHelper } from './discovery/locators/services/pipEnvServiceHelper';
+import { WindowsRegistryService } from './discovery/locators/services/windowsRegistryService';
+import { WindowsStoreInterpreter } from './discovery/locators/services/windowsStoreInterpreter';
+import {
+    WorkspaceVirtualEnvironmentsSearchPathProvider,
+    WorkspaceVirtualEnvService
+} from './discovery/locators/services/workspaceVirtualEnvService';
+import { WorkspaceVirtualEnvWatcherService } from './discovery/locators/services/workspaceVirtualEnvWatcherService';
+
+export function registerPythonEnvironmentsServices(serviceManager: IServiceManager) {
```

Comment:
Let's make it super obvious what this file is all about.  How about:

```suggestion
export function registerForIOC(serviceManager: IServiceManager) {
```

and call the file something like "legacyIOC.ts".